samples/modules/index.rst

```diff
@@ -4,7 +4,7 @@ External Module Samples
 #######################
 
 .. toctree::
-   :maxdepth: 2
+   :maxdepth: 3
```

Comment:
Did you verify that this doesn't produce too much noise in the [samples index](https://docs.zephyrproject.org/latest/samples/index.html)?

Reply:
I did not - I could move it back, but I think I would need to move stuff out of the `doc/` subdirectory to compensate, which is fine - I just need to then go to a few social media locations and update comments 😅

Reply:
@kartben do you have an opinion here?

Reply:
I think I'd rather get rid of the doc/ folder, frankly, as most samples tend to not have one and it's nice for the "See it on GitHub" link thingy to take one as close as possible to the actual source of a particular sample IMO.

That being said I can live with an update to the TOC depth, especially if you have existing links you don't want to break (in addition to pages from 3.3 which will 404 when people click on the "see the most recent version of this page banner).

Reply:
> (in addition to pages from 3.3 which will 404 when people click on the "see the most recent version of this page banner).

Are you sure? One problem this PR addresses is that this page is not present in the HTML for the toctree rooted here.

And anyway, that can be fixed with doc/_scripts/redirects.py if so.

Reply:
I thought Chris was saying he had social posts that may be directly linking to https://docs.zephyrproject.org/3.3.0/samples/modules/thrift/hello/doc/index.html -- page isn't mounted in the toc, but it's still there. So if we move it, yes, redirects will need to be updated (thanks for reminding me this exists :-))